lib/music_service/music_service_factory.dart

```diff
@@ -1,14 +1,14 @@
-import 'package:song_tinder/home_page/dummy_music_service.dart';
-import 'package:song_tinder/home_page/music_service_interface.dart';
-import 'package:song_tinder/home_page/song_provider.dart';
+import 'package:song_tinder/music_service/services/dummy_music_service.dart';
+import 'package:song_tinder/music_service/music_service_interface.dart';
+import 'package:song_tinder/providers/song_provider.dart';
+import 'package:song_tinder/music_service/services/spotify.dart';
 
 class MusicServiceFactory {
   static MusicServiceInterface from(MusicServiceConfig config) {
     print("Creating music service for ${config.service}");
     switch (config.service) {
       case MusicServices.spotify:
-        print("Spotify config: ID=${config.spotifyID}, ClientSecret=${config.spotifyClientSecret}");
-        break;
+        return Spotify();
```

Comment:
Why isn't config passed? Doesn't the Spotify object require the info in config to function?

Reply:
No it does not.

The clientId and clientSecret are hardcoded since they belong to my registered application on Spotify, the class does not require them on creation (instead it hold them as static members).